components/config/pref_util.rs

```diff
@@ -160,7 +159,7 @@ impl From<PrefValue> for [f64; 4] {
             PrefValue::Array(values) if values.len() == 4 => {
                 let f = values
                     .into_iter()
-                    .filter_map(|v| v.try_into().ok())
+                    .filter_map(|v| v.into())
```

Comment:
I believe this is revealing an issue with the surrounding code we've overlooked. This module implements `From<PrefValue> to f64`  - `From` trait represents an infallible conversion, but our impl panics when the conversion is not possible. This also means the logic in this particular method to check if all the elements are convertible is unnecessary as the code either panics when encountering the first failure or it succeeds for all i.e we can use `map` here instead of `filter_map` and we'll never reach the else case.

Not sure if this needs to be handled in this patch, but I guess we'll need to revisit the design here and maybe just implement TryFrom instead of panicking in From implementations.

Reply:
I agree, the move to `TryFrom` can be delayed.